pandas/core/arrays/base.py

```diff
@@ -762,7 +762,8 @@ def equals(self, other: "ExtensionArray") -> bool:
         """
         if not type(self) == type(other):
             return False
-        elif not self.dtype == other.dtype:
+        other = cast(ExtensionArray, other)
+        if not self.dtype == other.dtype:
```

Comment:
i guess this is fine, though more consisten to use is_dtype_equal no?

Reply:
I had to break the if elif to get the cast in. so this is unchanged. but agreed `is_dtype_equal` would be more consistent so will update.